gen_types.go

```diff
@@ -3562,6 +3562,20 @@ type Message struct {
 	ReplyMarkup *InlineKeyboardMarkup `json:"reply_markup,omitempty"`
 }
 
+// GetLink is a helper method to easily get the message link excluding group and private chat types.
+func (m *Message) GetLink() string {
+	if m.Chat.Type != "private" && m.Chat.Type != "group" {
+		var to_link string
```

Comment:
This variable adds an unnecessary extra layer of abstraction - it would be much simpler to use the previous "fail fast" approach and return the t.me string in two separate instances, one for public and the other for private supergroups, specifying two different `fmt.Sprintf` calls.

Also allows you to remove the extra indentation from the `else` statement. Win!

Reply:
Thanks for your valuable suggestions! I'll try to implement all the changes as much as I understood. :)